File: Semestrul 3/MAP/lab 2 MAP/decorator/DelayTaskRunner.java
package sem1_2.decorator;

public class DelayTaskRunner extends AbstractTaskRunner{

    public DelayTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask(){
        try{
            Thread.sleep(3000);
            super.executeOneTask();
        }
        catch (InterruptedException e){
            e.printStackTrace();
        }

    }

    @Override
    public void executeAll() {
        while (super.hasTask())
            executeOneTask();
    }
}

File: Semestrul 3/MAP/lab 10 MAP/decorator/DelayTaskRunner.cs
namespace seminar10.decorator;

public class DelayTaskRunner(ITaskRunner taskRunner) : AbstractTaskRunner(taskRunner)
{
    public override void ExecuteOneTask()
    {
        try
        {
            Thread.Sleep(3000);
            Console.WriteLine("Delay Task");
            base.ExecuteOneTask();
        }
        catch (Exception ex)
        {
            Console.WriteLine(ex.Message);
        }
    }

    public override void ExecuteAll()
    {
        while (HasTask())
        {
            ExecuteOneTask();
        }
    }
}